backend/inspirehep/search/serializers.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import orjson
+from elasticsearch import SerializationError
+from elasticsearch.compat import string_types
+from elasticsearch.serializer import JSONSerializer
+from flask import current_app
+
+from inspirehep.search.utils import RecursionLimit
+
+
+class ORJSONSerializerES(JSONSerializer):
+    def dumps(self, data):
+        if isinstance(data, string_types):
+            return data
+        try:
+            with RecursionLimit(
+                current_app.config.get("SEARCH_MAX_RECURSION_LIMIT", 5000)
+            ):
+                dump = orjson.dumps(data, default=self.default).decode("utf-8")
```

Comment:
Yes, they are, but there is additional code in es and in invenio to support dates and UUIDs and this requires them to be in a little different format than orjson will serialize/deserialize. We can optimize this later with proper testing how this will behave.

Reply:
Actually there would be much more optimization as now I have to decode dumped json, and then in ES library it's encoding it again just to get length of it :/ This whole part of ES we could remove as orjson produces dumps in this format already.

Reply:
And then it gets encoded again by the HTTP connection handler :( there's not much we can do about this though. For the `defaults` I would try without them though, as the `uuid` serialization in `orjson` is RFC-compliant, so it should be the same as what the `default` is doing.

Reply:
:+1: